rotkehlchen/utils/mixins/penalizable_oracle.py

```diff
@@ -19,7 +17,8 @@ def note_failure_or_penalize(self) -> None:
         It is called when an oracle fails, if the failure count reaches a threshold,
         it is punished for a duration i.e. skipped. Otherwise, the failure count is incremented.
         """
-        if self.query_failures_count >= ORACLE_PENALTY_THRESHOLD_COUNT:
+        settings = CachedSettings().get_settings()
+        if self.query_failures_count >= settings.oracle_penalty_threshold_count:
```

Comment:
instead of all the settings can't you just query the key?

```suggestion
        if self.query_failures_count >= CachedSettings().get_entry('oracle_penalty_threshold_count'):
```

Reply:
Okay, the reason for this was that the type checker wasn't pleased with the use of `get_entry`, because it returns
a union of types yet the one I need is an int.

It would therefore flag the comparison operations as invalid as they don't work with the non-integer types. But I could be missing a way of making it work in case you have a suggestion. Another approach would be to add a property method for each setting I need.

Reply:
As options you can:

- Create an overrade of the method so when the literal `oracle_penalty_threshold_count` is provided the return type is int.
- Use a `cast()`
- Use a `# type: ignore` explaining why

I would do the first or the last (probably the last and 1 in a smaller refactor) because the cast has a small overhead in production